does not work on two input fileGrps anymore

```
ocrd-pagetopdf -I OCR-D-OCR,OCR-D-BIN-WAN -O OCR-D-OUT -P textequiv_level word -P multipage merged --overwrite
Traceback (most recent call last): File "/data/ocr-d/ocrd_all/venv/bin/ocrd", line 8, in <module> sys.exit(cli())
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 1130, in __call__ return self.main(*args, **kwargs) File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 1055, in main rv = self.invoke(ctx)
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 1657, in invoke return _process_result(sub_ctx.command.invoke(sub_ctx)) 
File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 1657, in invoke return _process_result(sub_ctx.command.invoke(sub_ctx))
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 1404, in invoke return ctx.invoke(self.callback, **ctx.params)
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/click/core.py", line 760, in invoke return __callback(*args, **kwargs)
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/ocrd/cli/bashlib.py", line 115, in bashlib_input_files for input_file in processor.input_files:
 File "/data/ocr-d/ocrd_all/venv/lib/python3.7/site-packages/ocrd/processor/base.py", line 291, in input_files assert len(ret[0]) == 1, 'Use zip_input_files() instead of input_files when processing multiple input fileGrps'
 AssertionError: Use zip_input_files() instead of input_files when processing multiple input fileGrps 13:54:28.798
```

--- a/ocrd-pagetopdf
+++ b/ocrd-pagetopdf
@@ -11,29 +11,24 @@ SCRIPT_NAME="${0##*/}"
 
 MIMETYPE_PAGE=$(ocrd bashlib constants MIMETYPE_PAGE)
 
+FIFO=$(mktemp -u)
+function backout {
+    kill $(jobs -p)
+    wait &>/dev/null
+    exec 3>&-
+    rm -f $FIFO
+    exit 1
+}
+
 main () {
     # Load ocrd bashlib functions
     # shellcheck source=../core/ocrd/bashlib/lib.bash
     source $(ocrd bashlib filename)
     
-    ocrd__minversion 2.10.4
+    ocrd__minversion 2.50.0
 
     # Describe calling script to lib.bash
     ocrd__wrap "$SHAREDIR/ocrd-tool.json" "$SCRIPT_NAME" "$@"
-
-    cd "${ocrd__argv[working_dir]}"
-    local page_id in_file_grp img_file_grp out_file_grp
-    page_id=${ocrd__argv[page_id]:-}
-    mets="${ocrd__argv[mets_file]}"
-    IFS=',' read -ra in_grps <<< "${ocrd__argv[input_file_grp]}"
-    in_file_grp="${in_grps[0]}"
-    img_file_grp="${in_grps[1]:-}"
-    if [ -z "$img_file_grp" ]; then
-        ocrd log warning "Without a second input file group for images, the original imageFilename will be used"
-    fi
-    out_file_grp=${ocrd__argv[output_file_grp]}
-    
-    mkdir -p $out_file_grp
 
     # Parameters
     local negative2zero output_extension
@@ -60,36 +55,47 @@ main () {
     parameters+=(${params['script-args']})
     output_extension=${params['ext']}
 
+    cd "${ocrd__argv[working_dir]}"
+    local page_id in_file_grp img_file_grp out_file_grp
+    page_id=${ocrd__argv[page_id]:-}
+    mets="$(basename ${ocrd__argv[mets_file]})"
+    IFS=',' read -ra in_grps <<< "${ocrd__argv[input_file_grp]}"
+    in_file_grp="${in_grps[0]}"
+    img_file_grp="${in_grps[1]:-}"
+    if [ -z "$img_file_grp" ]; then
+        ocrd log warning "Without a second input file group for images, the original imageFilename will be used"
+    fi
+    out_file_grp=${ocrd__argv[output_file_grp]}
+    
+    mkfifo $FIFO
+    trap backout ERR
+    
+    bulk_options=( -r '(?P<grp>[^ ]+) (?P<page>[^ ]+) (?P<file>[^ ]+) (?P<url>.*)')
+    bulk_options+=( -G '{{ grp }}' -m application/pdf -g '{{ page }}' -i '{{ file }}' -S '{{ url }}')
+    if [[ "${ocrd__argv[overwrite]}" == true ]];then
+        bulk_options+=( --force )
+    fi
+    ocrd workspace -m "${mets}" bulk-add "${bulk_options[@]}" - <$FIFO &
+    exec 3>$FIFO
+    
     # Download the files and do the conversion
-    local IFS=$'\n'
-    files=($(ocrd workspace find \
-        ${page_id:+-g} ${page_id:-} \
-        -G $in_file_grp \
-        -k local_filename \
-        -k ID \
-        -k pageId \
-        --download))
-    local IFS=$' \t\n'
-    local n=0 zeros=0000
-    for csv in "${files[@]}"; do
-        let n+=1
-        # Parse comma separated fields
-        local IFS=$'\t'
-        local fields=($csv)
-        local IFS=$' \t\n'
-
-        local in_file="${fields[0]}"
-        local in_id="${fields[1]}"
-        local pageid="${fields[2]:-}"
-        declare -a options=(${parameters[*]})
-		
-        ocrd log info "processing page '$pageid'"
+    for ((n=0; n<${#ocrd__files[*]}; n++)); do
+        local in_file=($(ocrd__input_file $n url))
+        local in_id=($(ocrd__input_file $n ID))
+        local in_mimetype=($(ocrd__input_file $n mimetype))
+        local in_pageId=($(ocrd__input_file $n pageId))
+        local out_id="$(ocrd__input_file $n outputFileId)"
+        local out_file="$out_file_grp/${out_id}.xml"
         
-        if ! test -f "$in_file"; then
-           ocrd log error "input file \"$in_file\" ID=${in_id} (pageId=${pageid}) is not on disk"
+        declare -a options=(${parameters[*]})
+        ocrd log info "processing page '${in_pageId}'"
+        
+        if ! test -f "${in_file#file://}"; then
+           ocrd log error "input file \"${in_file#file://}\" (ID=${in_id} pageId=${in_pageId} MIME=${in_mimetype}) is not on disk"
            continue
         fi
-        ocrd log info "found PAGE file '$in_file'"
+        mkdir -p $out_file_grp
+        ocrd log info "found ${in_mimetype} file '${in_file}'"
         
         # Rework coords in PAGE
         if ((negative2zero)); then
@@ -101,18 +107,17 @@ main () {
 	    options+=(-xml "$in_file")
 
         if [ -n "$img_file_grp" ]; then
-            img_file=$(ocrd workspace find \
-                            -g $pageid -G $img_file_grp \
-                            -m '//image/.*' \
-                            -k local_filename --download)
+            # multiple fileGrps: ocrd__input_file() is n-ary
+            img_file="${in_file[1]}"
         else
-            # we could use xsltproc or xmlstarlet for this
-            # (but that would add another dependency)
-            img_file=$(python3 "$SHAREDIR/ptp/extract-imagefilename.py" "$in_file")
+            img_file="${in_file}"
         fi
+        # we could use xsltproc or xmlstarlet for this
+        # (but that would add another dependency)
+        img_file=$(python3 "$SHAREDIR/ptp/extract-imagefilename.py" "$in_file")
 
         if ! test -f "$img_file"; then
-           ocrd log error "No image file '$img_file' for $in_id (pageId=$pageid)"
+           ocrd log error "No image file '$img_file' for $in_id (pageId=$in_pageId)"
            continue
         fi
         options+=(-image "$img_file")
@@ -128,10 +133,10 @@ main () {
  
         ocrd log info "found image file '$img_file'"
         if ! output=$(java -jar "$SHAREDIR/ptp/PageToPdf.jar" "${options[@]}" 2>&1); then
-            ocrd log error "PageToPdf failed for ID $in_id (pageId=$pageid): $output"
+            ocrd log error "PageToPdf failed for ID $in_id (pageId=$in_pageId): $output"
             continue
         elif ! [ -f "$out_file" -a -s "$out_file" ]; then
-            ocrd log error "PageToPdf result is empty for ID $in_id (pageId=$pageid): $output"
+            ocrd log error "PageToPdf result is empty for ID $in_id (pageId=$in_pageId): $output"
             continue
         fi
         
@@ -139,23 +144,26 @@ main () {
             rm $tmpfile
         fi
 
-        # Add the output file to METS
-        if [ -n "$pageid" ]; then
-            options=( -g $pageid )
-        else
-            options=()
-        fi
-        if [[ "${ocrd__argv[overwrite]}" = true ]]; then
-            options+=( --force )
-        fi
-        options+=( -G $out_file_grp
-                   -m application/pdf
-                   -i "$out_id"
-                   "$out_file" )
         ocrd log info "adding output PDF file '$out_file'"
-        ocrd workspace add "${options[@]}"
-
+        # Add the output file to METS
+        # if [ -n "$in_pageId" ]; then
+        #     options=( -g $in_pageId )
+        # else
+        #     options=()
+        # fi
+        # if [[ "${ocrd__argv[overwrite]}" = true ]]; then
+        #     options+=( --force )
+        # fi
+        # options+=( -G $out_file_grp
+        #            -m application/pdf
+        #            -i "$out_id"
+        #            "$out_file" )
+        # ocrd workspace add "${options[@]}"
+        echo "${out_file_grp}" "${in_pageId}" "${out_id}" "${out_file}" >&3
     done
+    exec 3>&-
+    rm -f $FIFO
+    wait
     
     if [ ${params['multipage']:=} ]; then
         python3 "$SHAREDIR/ptp/multipagepdf.py" "$mets" "$out_file_grp" "$page_id" "${params['multipage']}" "${params['pagelabel']}" "${ocrd__argv['overwrite']}"

--- a/ocrd-pagetopdf
+++ b/ocrd-pagetopdf
@@ -97,6 +97,15 @@ main () {
         mkdir -p $out_file_grp
         ocrd log info "found ${in_mimetype} file '${in_file}'"
         
+        if [ -n "$img_file_grp" ]; then
+            # multiple fileGrps: ocrd__input_file() is n-ary
+            img_file="${in_file[1]}"
+            img_mime="${in_mimetype[1]}"
+        else
+            img_file="${in_file}"
+            img_mime="${in_mimetype}"
+        fi
+
         # Rework coords in PAGE
         if ((negative2zero)); then
             local tmpfile
@@ -106,15 +115,11 @@ main () {
         fi
 	    options+=(-xml "$in_file")
 
-        if [ -n "$img_file_grp" ]; then
-            # multiple fileGrps: ocrd__input_file() is n-ary
-            img_file="${in_file[1]}"
-        else
-            img_file="${in_file}"
+        if [ "$img_mime" = "$MIMETYPE_PAGE" ]; then
+            # we could use xsltproc or xmlstarlet for this
+            # (but that would add another dependency)
+            img_file=$(python3 "$SHAREDIR/ptp/extract-imagefilename.py" "$img_file")
         fi
-        # we could use xsltproc or xmlstarlet for this
-        # (but that would add another dependency)
-        img_file=$(python3 "$SHAREDIR/ptp/extract-imagefilename.py" "$in_file")
 
         if ! test -f "$img_file"; then
            ocrd log error "No image file '$img_file' for $in_id (pageId=$in_pageId)"
@@ -130,7 +135,6 @@ main () {
         local out_file="$out_file_grp/${out_id}$output_extension"
         options+=(-pdf "$out_file")
         
- 
         ocrd log info "found image file '$img_file'"
         if ! output=$(java -jar "$SHAREDIR/ptp/PageToPdf.jar" "${options[@]}" 2>&1); then
             ocrd log error "PageToPdf failed for ID $in_id (pageId=$in_pageId): $output"

--- a/ptp/multipagepdf.py
+++ b/ptp/multipagepdf.py
@@ -6,7 +6,7 @@ import subprocess
 
 from ocrd_models import OcrdMets
 from ocrd_utils.logging import getLogger, initLogging
-from atomicwrites import atomic_write
+from ocrd_utils import atomic_write
 from ocrd_models.constants import NAMESPACES as NS
 
 def get_metadata(mets):
@@ -40,7 +40,7 @@ def read_from_mets(metsfile, filegrp, page_ids, outputfile, pagelabel='pageId', 
         mets.add_file(filegrp, mimetype='application/pdf', ID=outputfile,
                       url=str(Path(filegrp).joinpath(outputfile+'.pdf')),
                       force=overwrite)
-        with atomic_write(metsfile, overwrite=True) as f:
+        with atomic_write(metsfile) as f:
             f.write(mets.to_xml(xmllint=True).decode('utf-8'))
 
 def create_pdfmarks(pdfdir, pagelabels=None, metadata=None):

